### ISSUE TITLE
Build error - certificate validation

### ISSUE THREAD

**Issue by pstephenson02** [tozny/e3db#36]
Hello! I'm interested in playing around with e3db, but I'm having problems registering. I've downloaded the CLI tool and when I try to register:

```bash
phil@philbuntu:~/e3db-0.7.1/bin$ ./e3db register
E-Mail Address []: phil.stephenson@gmail.com
com.tozny.e3db.client.errors.IOError: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at com.tozny.e3db.client.Registration.rawCall(Registration.java:64)
	at com.tozny.e3db.client.Registration.register(Registration.java:131)
	at com.tozny.e3db.cli.Main$.do_register(main.scala:92)
	at com.tozny.e3db.cli.Main$.runWithOpts(main.scala:391)
	at com.tozny.e3db.cli.Main$.liftedTree1$1(main.scala:410)
	at com.tozny.e3db.cli.Main$.main(main.scala:409)
	at com.tozny.e3db.cli.Main.main(main.scala)
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)
	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)
	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296)
	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1509)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:914)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:241)
	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:198)
	at okhttp3.internal.connection.RealConnection.buildConnection(RealConnection.java:174)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:114)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:193)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:129)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:98)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:109)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:124)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:170)
	at okhttp3.RealCall.execute(RealCall.java:60)
	at com.tozny.e3db.client.Registration.rawCall(Registration.java:54)
	... 6 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387)
	at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:230)
	at sun.security.validator.Validator.validate(Validator.java:260)
	at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1491)
	... 34 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
	at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
	at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)
	at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382)
	... 40 more
```

I've also tried to build from source, but saw similar errors in [my build log](https://gist.github.com/pstephenson02/fefc2484d700b97b5ad9bb633c1b78de). Also tried building with an older version of sbt via `./sbt sbt-version 0.13.1` which got me past the cert errors but I ended up in a bad state as well.

I'm on Ubuntu 16.04.1 LTS - apparently this is a common issue with sbt on Debian: https://github.com/sbt/sbt/issues/2536
I tried [this person's workaround](https://github.com/sbt/sbt/issues/2536#issuecomment-243997724) but it did not work for me. I started with OpenJDK 8 and then installed Oracle Java8.

I realize this issue likely has little, if anything to do with your e3db cli, but I'm wondering if you have any advice on getting this resolved? I could also setup a virtual machine on a specific platform if you know one that definitely works. Thanks in advance!

**Comment by SyntaxPolice** [comment]
Thanks for the bug report! My understanding of the bug is this: This doesn't look like an sbt issue since the CLI comes pre-built; it's able to run the command, but when it talks to our server, it's not able to create a secure connection because the root certificate we use is not in your java trust store for some reason. I just double-checked our certs and they look good as far as I can tell.

Assuming no one (including some proxy? or hotel wifi?) is man-in-the-middling your connection, it may be that Ubuntu's Java doesn't trust various certs by default? One *guess* is that these packages install the "standard" root certs:
apt-get install ca-certificates-java ca-certificates

Alternately, you can install the godaddy root cert by hand in the java keychain. Let me know if any of this info helps. We have tested on Ubuntu, so I'm surprised it's a problem. I'd be interested to know what the problem is since others on ubuntu are likely to have the same problem.

**Comment by pstephenson02** [comment]
Thanks for the quick response @SyntaxPolice. The ca-certificates packages I already had installed on my system, but your note about the root cert missing from my trust store was helpful! I looked at that and could not find the godaddy root cert. Rather than installing it manually, I ended up just uninstalling both openjdk and oracle java, removing lingering configuration and then re-installing. After doing that and checking again, the number of certs in my store had increased and I could see the go daddy root cert. At this point I was able to register!

I suspect that I had installed java a long while ago, possibly incorrectly or changed something somewhere and I was just missing one (or more) of those root certificates. I know this is not a very helpful or reproducible bug report, but I'll close out this issue now. Cheers!
